iotdb/src/main/java/org/apache/iotdb/db/engine/bufferwrite/BufferWriteProcessor.java

```diff
@@ -500,4 +506,20 @@ public void setNewProcessor(boolean isNewProcessor) {
   public WriteLogNode getLogNode() {
     return logNode;
   }
+
+  /**
+   * Delete data whose timestamp <= 'timestamp' and belonging to timeseries deviceId.measurementId.
+   * Delete data in both working MemTable and flushing MemTable.
+   * @param deviceId the deviceId of the timeseries to be deleted.
+   * @param measurementId the measurementId of the timeseries to be deleted.
+   * @param timestamp the upper-bound of deletion time.
+   */
+  public void delete(String deviceId, String measurementId, long timestamp) {
+    workMemTable.delele(deviceId, measurementId, timestamp);
+    if (isFlush) {
+      // flushing MemTable cannot be directly modified since another thread is reading it
+      flushMemTable = flushMemTable.copy();
+      flushMemTable.delele(deviceId, measurementId, timestamp);
```

Comment:
Hm... so is this operation is just for guaranteeing `queryBufferWriteData()` ? 
Actually, `flushMemTable = flushMemTable.copy();` can not affect the flushing memtable... so you have to add the Deletion operation in the modification file at the same time.

Reply:
Exactly.